core-backend/sandbox/src/env.rs

```diff
@@ -159,14 +160,18 @@ impl<E: Ext + Into<ExtInfo> + 'static> Environment<E> for SandboxEnvironment<E>
         let info: ExtInfo = runtime.ext.unset().into();
 
         let termination = if res.is_err() {
-            let mut reason = None;
-
-            if let Some(trap) = runtime.trap {
-                if common_funcs::is_wait_trap(trap) {
-                    reason = Some(TerminationReason::Manual { wait: true });
+            let reason = if let Some(trap) = runtime.trap {
```

Comment:
Can't we somehow simplify this matching? @shamilsan

Reply:
We could write:

```rust
let reason = match runtime.trap {
    ...
    Some(trap) if common_funcs::is_wait_trap(trap) => Some(TerminationReason::Wait),
    Some(trap) if common_funcs::is_leave_trap(trap) => Some(TerminationReason::Leave),
    ...
    _ => None,
```

But I don't think it's much better :)